RavenDB-15036 : Query should be able to use time-series value name

aviv86 opened the pull request:
https://issues.hibernatingrhinos.com/issue/RavenDB-15036

aviv86 committed:
RavenDB-15036 : Query should be able to use time-series value name

aviv86 committed:
RavenDB-15036 : tests

ppekrol commented on src/Raven.Server/Documents/Queries/Results/TimeSeriesRetriever.cs:
```diff
@@ -724,6 +724,55 @@ string GetSourceAndId()
+                using (txContext.OpenReadTransaction())
+                using (var dbDoc = _context.DocumentDatabase.ServerStore.Cluster.Read(txContext, dbId, out long etag))
+                {
+                    _databaseRecord = DocumentConventions.Default.Serialization.DefaultConverter.FromBlittable<DatabaseRecordWithEtag>(dbDoc, "database/record");
```
Use `ReadRawDatabaseRecord`. Here you are materializing the whole record

karmeli87 replied:
use `ReadRawDatabaseRecord` and then `TimeSeriesConfiguration`,  we don't need the entire database record here

karmeli87 commented on src/Raven.Server/Documents/Queries/Results/TimeSeriesRetriever.cs:
```diff
@@ -724,6 +724,55 @@ string GetSourceAndId()
+                }
+            }
+
+            if (_databaseRecord.TimeSeries?.NamedValues == null ||
+                _databaseRecord.TimeSeries.NamedValues.TryGetValue(_collection, out var dictionary) == false ||
+                dictionary.TryGetValue(_source, out var namedValues) == false)
+                return null;
+            
+            int index;
+            for (index = 0; index < namedValues.Length; index++)
+            {
+                if (namedValues[index] == fieldValue)
+                    break;
+            }
+
+            if (index == namedValues.Length ||
+                index >= singleResult.Values.Length) // shouldn't happen
+                return null;
+
```
you can use GetNames

aviv86 committed:
RavenDB-15036 : address PR comments

aviv86 committed:
RavenDB-15036 : minor changes :
- GetNamedValue : use ReadTimeSeriesConfiguration instead of ReadRawDatabaseRecord,
- AddNamesIfNeeded : use _namedValues
